curvefs/src/client/fuse_client.cpp

```diff
@@ -1025,16 +1113,88 @@ CURVEFS_ERROR FuseClient::CalAllLayerSumInfo(Inode *inode) {
+    std::atomic<bool> *ret) {
+    while (1) {
+        std::list<Dentry> dentryList;
+        std::set<uint64_t> inodeIds;
+        std::list<XAttr> xattrs;
+        auto tret = ConcurrentListDentry(&dentryList, iStack, stackMutex,
+                                         dirOnly, inflightNum, ret);
+        if (!tret) {
+            break;
+        } else {
+            {
+                std::lock_guard<std::mutex> guard(*stackMutex);
+                for (const auto &it : dentryList) {
+                    iStack->emplace(it.inodeid());
+                    inodeIds.emplace(it.inodeid());
+                }
+            }
+            inflightNum->fetch_sub(1, std::memory_order_seq_cst);
+            if (!inodeIds.empty()) {
+                auto tret = inodeManager_->BatchGetXAttr(&inodeIds, &xattrs);
+                if (tret == CURVEFS_ERROR::OK) {
+                    std::lock_guard<std::mutex> guard(*inodeMutex);
+                    for (const auto &it : xattrs) {
+                        if (it.xattrinfos().count(XATTRFILES)) {
+                            if (!AddUllStringToFirst(
+                                &(inode->mutable_xattr()->find(XATTRRFILES)->second),    // NOLINT
+                                it.xattrinfos().find(XATTRFILES)->second,
+                                true)) {
+                                ret->exchange(false);
+                                break;
+                            }
+                        }
+                        if (it.xattrinfos().count(XATTRSUBDIRS)) {
+                            if (!AddUllStringToFirst(
+                                &(inode->mutable_xattr()->find(XATTRRSUBDIRS)->second),    // NOLINT
+                                it.xattrinfos().find(XATTRSUBDIRS)->second,
+                                true)) {
+                                ret->exchange(false);
+                                break;
+                            }
+                        }
+                        if (it.xattrinfos().count(XATTRENTRIES)) {
+                            if (!AddUllStringToFirst(
+                                &(inode->mutable_xattr()->find(XATTRRENTRIES)->second),    // NOLINT
+                                it.xattrinfos().find(XATTRENTRIES)->second,
+                                true)) {
+                                ret->exchange(false);
+                                break;
+                            }
+                        }
+                        if (it.xattrinfos().count(XATTRFBYTES)) {
+                            if (!AddUllStringToFirst(
+                                &(inode->mutable_xattr()->find(XATTRRFBYTES)->second),    // NOLINT
+                                it.xattrinfos().find(XATTRFBYTES)->second,
+                                true)) {
+                                ret->exchange(false);
+                                break;
+                            }
+                        }
+                    }
```

Comment:
you add all subdirs/subfiles information into the target inode, for the calculations here, you use string addition and subtraction.

and I think you can just use integrals to sum the middle values, and finally, convert the value to string and assign the result to inode.

Reply:
fixed